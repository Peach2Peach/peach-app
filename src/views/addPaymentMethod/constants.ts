import { CurrencyType } from "../../store/offerPreferenes/types";

export const defaultCurrencies: Record<CurrencyType, Currency> = {
  europe: "EUR",
  latinAmerica: "ARS",
  africa: "ZAR",
  global: "USDT",
  asia: "AED",
  northAmerica: "CAD",
  middleEast: "ILS",
  oceania: "AUD",
};
