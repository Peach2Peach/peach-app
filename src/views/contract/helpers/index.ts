export { isPSBTForBatch } from "./isPSBTForBatch";
export { shouldShowTradeStatusInfo } from "./shouldShowTradeStatusInfo";
export { verifyReleasePSBT as verifyPSBT } from "./verifyReleasePSBT";
