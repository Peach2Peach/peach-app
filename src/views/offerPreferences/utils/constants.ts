const horizontalPaddingForSlider = 8
export const trackMin = horizontalPaddingForSlider
