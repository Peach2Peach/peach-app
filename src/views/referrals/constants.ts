export const POINTS_PER_FREE_TRADE = 100;
export const NUMBER_OF_FREE_TRADES = 2;
export const REWARDINFO: Reward[] = [
  { id: "customReferralCode", requiredPoints: 100 },
  {
    id: "noPeachFees",
    requiredPoints: POINTS_PER_FREE_TRADE * NUMBER_OF_FREE_TRADES,
  },
  { id: "sats", requiredPoints: 300 },
];
