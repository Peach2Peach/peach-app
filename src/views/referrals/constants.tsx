export const REWARDINFO: Reward[] = [
  { id: 'customReferralCode', requiredPoints: 100 },
  { id: 'noPeachFees', requiredPoints: 200 },
  { id: 'sats', requiredPoints: 300 },
]
