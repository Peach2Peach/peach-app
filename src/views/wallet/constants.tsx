export const WALLETS: Record<Chain, keyof HomeTabParamList> = {
  bitcoin: "wallet",
  lightning: "lightningWallet",
  liquid: "liquidWallet",
};
