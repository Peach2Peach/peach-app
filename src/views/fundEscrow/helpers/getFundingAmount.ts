import { FundMultipleInfo } from '../../../utils/wallet/walletStore'

export const getFundingAmount = (fundMultiple?: FundMultipleInfo, amount = 0) =>
  fundMultiple ? amount * fundMultiple.offerIds.length : amount
