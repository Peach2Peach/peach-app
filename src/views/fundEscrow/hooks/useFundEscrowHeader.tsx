import { networks } from 'bitcoinjs-lib'
import { useCancelOffer, useHeaderSetup } from '../../../hooks'
import { useCancelFundMultipleSellOffers } from '../../../hooks/useCancelFundMultipleSellOffers'
import { useShowHelp } from '../../../hooks/useShowHelp'
import i18n from '../../../utils/i18n'
import { headerIcons } from '../../../utils/layout/headerIcons'
import { generateBlock } from '../../../utils/regtest'
import { getNetwork } from '../../../utils/wallet'
import { FundMultipleInfo } from '../../../utils/wallet/walletStore'

type Props = {
  fundingStatus: FundingStatus
  sellOffer?: SellOffer
  fundMultiple?: FundMultipleInfo
}
export const useFundEscrowHeader = ({ fundingStatus, sellOffer, fundMultiple }: Props) => {
  const showHelp = useShowHelp('escrow')
  const showMempoolHelp = useShowHelp('mempool')
  const cancelOffer = useCancelOffer(sellOffer)
  const cancelFundMultipleOffers = useCancelFundMultipleSellOffers({ fundMultiple })

  const headerConfig
    = fundingStatus.status === 'MEMPOOL'
      ? {
        title: i18n('sell.funding.mempool.title'),
        icons: [{ ...headerIcons.help, onPress: showMempoolHelp }],
      }
      : {
        title: i18n('sell.escrow.title'),
        icons: [
          { ...headerIcons.cancel, onPress: fundMultiple ? cancelFundMultipleOffers : cancelOffer },
          { ...headerIcons.help, onPress: showHelp },
        ],
      }

  if (getNetwork() === networks.regtest) {
    headerConfig.icons.unshift({ ...headerIcons.generateBlock, onPress: generateBlock })
  }
  useHeaderSetup(headerConfig)
}
