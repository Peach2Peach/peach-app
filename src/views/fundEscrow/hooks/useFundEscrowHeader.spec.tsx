import { renderHook } from '@testing-library/react-native'
import { account1 } from '../../../../tests/unit/data/accountData'
import { sellOffer } from '../../../../tests/unit/data/offerData'
import { NavigationWrapper, headerState, setOptionsMock } from '../../../../tests/unit/helpers/NavigationWrapper'
import { updateAccount } from '../../../utils/account'
import { useFundEscrowHeader } from './useFundEscrowHeader'
import { defaultFundingStatus } from '../../../utils/offer/constants'

const showHelpMock = jest.fn()
const useShowHelpMock = jest.fn((..._args) => showHelpMock)
jest.mock('../../../hooks/useShowHelp', () => ({
  useShowHelp: (...args: any) => useShowHelpMock(...args),
}))

const wrapper = NavigationWrapper

const cancelOfferMock = jest.fn()
jest.mock('../../../hooks/useCancelOffer', () => ({
  useCancelOffer: () => cancelOfferMock,
}))

describe('useFundEscrowHeader', () => {
  beforeEach(() => {
    setOptionsMock({ header: { title: '', icons: [] } })

    updateAccount({ ...account1, offers: [] }, true)
  })

  it('should render header correctly for unfunded offers', () => {
    renderHook(useFundEscrowHeader, {
      wrapper,
      initialProps: {
        fundingStatus: defaultFundingStatus,
        sellOffer,
      },
    })
    expect(headerState.header()).toMatchSnapshot()
    expect(headerState.header().props.icons?.[0].onPress).toEqual(cancelOfferMock)
    expect(headerState.header().props.icons?.[1].onPress).toEqual(showHelpMock)
  })
  it('should render header correctly for funding in mempool', () => {
    renderHook(useFundEscrowHeader, {
      wrapper,
      initialProps: {
        fundingStatus: { ...defaultFundingStatus, status: 'MEMPOOL' },
        sellOffer,
      },
    })
    expect(headerState.header()).toMatchSnapshot()
    expect(headerState.header().props.icons?.[0].onPress).toEqual(showHelpMock)
  })
})
