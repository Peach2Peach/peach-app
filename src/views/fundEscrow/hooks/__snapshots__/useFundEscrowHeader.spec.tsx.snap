// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useFundEscrowHeader should render header correctly for funding in mempool 1`] = `
<Header
  icons={
    [
      {
        "color": "#099DE2",
        "id": "helpCircle",
        "onPress": [MockFunction],
      },
    ]
  }
  title="transaction pending"
/>
`;

exports[`useFundEscrowHeader should render header correctly for unfunded offers 1`] = `
<Header
  icons={
    [
      {
        "color": "#9F8C82",
        "id": "xCircle",
        "onPress": [MockFunction],
      },
      {
        "color": "#099DE2",
        "id": "helpCircle",
        "onPress": [MockFunction],
      },
    ]
  }
  title="fund escrow"
/>
`;

exports[`useFundEscrowHeader should show regtest icons 1`] = `
<Header
  icons={
    [
      {
        "color": "#F5CE22",
        "id": "cpu",
        "onPress": [Function],
      },
      {
        "color": "#9F8C82",
        "id": "xCircle",
        "onPress": [MockFunction],
      },
      {
        "color": "#099DE2",
        "id": "helpCircle",
        "onPress": [MockFunction],
      },
    ]
  }
  title="fund escrow"
/>
`;
